docs/nachet-model-documentation.md

```diff
@@ -0,0 +1,190 @@
+|[Segmentation](https://huggingface.co/tasks/image-segmentation) | Segmentation is the task of dividing images into different parts, where each pixel in the image is mapped to an object. It includes instance segmentation, panoptic segmentation, and semantic segmentation.| The segmentation models take an image as input and return an image divided into objects. |
+
+> As of today (2024-02-22), no model uses segmentation. To know more about each
+> task, click on the task to follow the link to their hugging face page. To know
+> more about AI tasks in general: [Hugging Face
+> Tasks](https://huggingface.co/tasks)
+
+## List of models
+
+|Model|Full name|Task|API Call Function|Inference Function|Active|Accuracy|
+|--|--|:--:|:--:|:--:|:--:|:--:|
+|Nachet-6seeds | m-14of15seeds-6seedsmag | Object Detection | nachet_6seeds | None | Yes | - |
+|Seed-detector | seed-detector-1 |  Object Detection | seed_detector | process_image_slicing | Yes | - |
+|Swin | swinv1-base-dataaugv2-1 | Classification | swin | process_swin_result | Yes | - |
+
+### Inference and API Call Function
+
+The inference and API call functions act as entry and exit points for the model. The API call explicitly requests a prediction from the specified model (such as Swin, Nachet-6seeds, etc.). The inference function processes the data before sending it to the frontend if the model requires it. For instance, the Seed-detector only returns "seed" as a label, and its inference needs to be processed and passed to the next model which assigns the correct label to the seeds.
+
+## Return value of models
+
+```json
+{
+    "filename": "tmp/tmp_file_name",
+    "boxes": [
+        {"box": {
+                "topX": 0.0,
+                "topY": 0.0,
+                "bottomX": 0.0,
+                "bottomY": 0.0
+            },
+        "label": "top_label_name",
+        "score": 0.912,
+        "topN": [
+            {
+                "score": 0.912,
+                "label": "top_label_name",
+            },
+            {
+                "score": 0.053,
+                "label": "seed_name",
+            },
+            {
+                "score": 0.0029,
+                "label": "seed_name",
+            },
+            {
+                "score": 0.005,
+                "label": "seed_name",
+            },
+            {
+                "score": 0.001,
+                "label": "seed_name",
+            }
+        ],
+        "overlapping": false,
+        "overlappingIndices": 0
+        },
+    ],
+    "labelOccurrence": {
+        "seed_name": 1,
+    },
+    "totalBoxes": 1
+}
+```
+
+### Why topN
+
+We decided to named the top results property top N because this value can return n predictions. Usually in AI, the top 5 result are use to measure the accuracy of a model. If the correct result is the top 5, then it is considered that the prediction was true.
```

Comment:
Usually in AI, the top 5 result**s** are use**d** to measure the accuracy of a model. If the correct result is **in** the top 5, then the prediction is considered true.

Reply:
Done